CMakeLists.txt

```diff
@@ -44,7 +44,7 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
   set(CMAKE_CXX_EXTENSIONS OFF)
 
   if(UNIX)
-    SET(COMPILE_FLAGS "-g -ggdb -O0")
+    SET(COMPILE_FLAGS "-g -ggdb -O0 -fno-rtti")
```

Comment:
Is there any specific reason of adding the `-fno-rtti` flag? 

From https://gcc.gnu.org/onlinedocs/gcc-4.4.7/gcc/C_002b_002b-Dialect-Options.html
> -fno-rtti
>     Disable generation of information about every class with virtual functions for use by the C++ runtime type identification features (`dynamic_cast` and `typeid`). If you don't use those parts of the language, you can save some space by using this flag. Note that exception handling uses the same information, but it will generate it as needed. The `dynamic_cast' operator can still be used for casts that do not require runtime type information, i.e. casts to void * or to unambiguous base classes. 
```